nav2_collision_monitor/src/polygon.cpp

```diff
@@ -562,4 +543,48 @@ inline bool Polygon::isPointInside(const Point & point) const
   return res;
 }
 
+bool Polygon::getPolygonFromString(
+  std::string &  poly_string,
+  std::vector<Point> & polygon)
+{
+  std::string error;
+  std::vector<std::vector<float>> vvf = nav2_util::parseVVF(poly_string, error);
+
+  if (error != "") {
+    RCLCPP_ERROR(
+      rclcpp::get_logger(
+        "nav2_collision_monitor"), "Error parsing polygon parameter: '%s'", error.c_str());
+    RCLCPP_ERROR(
+      rclcpp::get_logger(
+        "nav2_collision_monitor"), "  Polygon string was '%s'.", poly_string.c_str());
+    return false;
+  }
+
+  // Check for minimum 3 points
+  if (vvf.size() < 3) {
+    RCLCPP_ERROR(
+      rclcpp::get_logger(
+        "nav2_collision_monitor"),
+      "Polygon must have at least three points.");
+    return false;
+  }
+  for (unsigned int i = 0; i < vvf.size(); i++) {
+    if (vvf[i].size() == 2) {
+      Point point;
+      point.x = vvf[i][0];
+      point.y = vvf[i][1];
+      polygon.push_back(point);
+    } else {
+      RCLCPP_ERROR(
+        rclcpp::get_logger(
```

Comment:
`RCLCPP_ERROR(logger_, "")` You have a `logger_` object you can use. Update here and in above warnings too 

Also, don't `NOLINT` it, use line extensions in the string if its > 100 chars